beacon_node/execution_layer/src/lib.rs

```diff
@@ -367,8 +405,8 @@ impl<T: EthSpec> ExecutionLayer<T> {
     }
 
     /// Cache a full payload, keyed on the `tree_hash_root` of its `transactions` field.
-    fn cache_payload(&self, payload: &ExecutionPayload<T>) -> Option<ExecutionPayload<T>> {
-        self.inner.payload_cache.put(payload.clone())
+    fn cache_payload(&self, payload: ExecutionPayloadRef<T>) -> Option<ExecutionPayload<T>> {
+        self.inner.payload_cache.put(payload.clone_from_ref())
     }
 
     /// Attempt to retrieve a full payload from the payload cache by the `transactions_root`.
```

Comment:
I believe these comments are incorrect. Isn't the cache keyed by the `tree_hash_root` of the entire header and not just the `transactions`?

Reply:
Yeah good catch, may as well update this while we notice it